### DISCUSSION TITLE
Set Hierarchy for Renovatebot updates

### DISCUSSION THREAD

**Discussion by MeNsaaH** [renovatebot/renovate#18981] [Request Help] [answered]
I have a repository container kubernetes manifests deployed using Argocd. I plan to use Renovate bot to bump versions of images, helm charts and kustomize refs. 
```json
{
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "additionalBranchPrefix": "{{baseDir}}-",
      "matchPackagePatterns": ["*"],
      "matchUpdateTypes": ["major", "minor"]
    }
  ],
  "patch": {
    "enabled": false
  },
  "ignoreDeps": ["null"],
  "schedule": ["before 8am every weekday"],
  "prHourlyLimit": 10,
  "commitMessagePrefix": "{{parentDir}}: "
}
```
With the config above, Renovate sends different PRs for staging and production directories because of the way my repo is structured. I want renovatebot to send PRs for staging first before sending PRs for production. However, when the PRs come in, it's always the production PRs that come in first. Is there anyway I can set like a hierarchy for pullrequests to make renovatebot create PRs for stagings first before Production

**Comment by rarkins** [comment, marked as answer]
Renovate by default will prioritise patch then minor then major updates. If you're getting production first it's likely because production has more minor updates, or simply bad luck due to which updates get sorted first when all else is equal.

You could add a package rule which uses matchPaths + prPriority to either increase staging priority or decrease production priority.

BTW disabling patch updates seems a little strange to me, are you sure about that?

**Reply by MeNsaaH** [reply]
@rarkins We use mostly helm charts. Most of the patch changes introduced are just changes that are not critical.  Maybe, I'll tune this to disable patch updates only on helm charts. Is there a way to do this easily?

I'm testing out the `matchPaths` + `prPriority`. I'll let you know how that goes.

**Reply by rarkins** [reply]
> @rarkins We use mostly helm charts. Most of the patch changes introduced are just changes that are not critical. Maybe, I'll tune this to disable patch updates only on helm charts. Is there a way to do this easily?

Yes, use a packageRule with matchDatasources=[helm], matchUpdateTypes=[patch], enabled=false

**Reply by MeNsaaH** [reply]
Thanks @rarkins . The `prPriority` did the trick for staging and production changes 🙇🏿 . Thanks for your answers and help. I'll also enable patch and disable it just for helm
